src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -936,24 +988,25 @@ private static bool IsViableType(LookupResult result)
         /// </summary>
         private NamedTypeSymbol BindNativeIntegerSymbolIfAny(IdentifierNameSyntax node, BindingDiagnosticBag diagnostics)
         {
-            SpecialType specialType;
-            switch (node.Identifier.Text)
-            {
-                case "nint":
-                    specialType = SpecialType.System_IntPtr;
-                    break;
-                case "nuint":
-                    specialType = SpecialType.System_UIntPtr;
-                    break;
-                default:
```

Comment:
Consider reverting this change if this is a style change only. It looks like the previous code may require fewer calls.

Reply:
i think the new form is preferred.  Instead of hardcoding in the strings, we just defer to the fact that the node knows if it is nint/nuint.